packages/playground-examples/copy/ja/TypeScript/Meta-Types/Conditional Types.ts

```diff
@@ -0,0 +1,109 @@
+// 条件付き型はTypeScriptの型システムに
+// 簡単なロジックを組み込む方法を提供します。
+// これは非常に高度な機能のため、
+// 日々の開発において使わないことも十分に可能です。
+
+// 条件付き型は以下のような形です:
+//
+//   A extends B ? C : D
+//
+// 条件部はある型がある式を拡張しているかを定義し、
+// 条件を満たす場合にどの型を返すかを定義します。
+
+// いくつかの例を見てみましょう。
+// ここでは簡潔さのためにジェネリクスに単一の文字を使用していきます。
+// これは任意ですが、
+// 60文字に制限すると画面に収まり難くなります。
+
+type Cat = { meows: true };
+type Dog = { barks: true };
+type Cheetah = { meows: true; fast: true };
+type Wolf = { barks: true; howls: true };
+
+// 以下のように吠える(barks)動物の型だけを
+// 抜き出す条件付き型を作れます。
+
+type ExtractDogish<A> = A extends { barks: true } ? A : never;
+
+// 次に、ExtractDogishで包んだ型を作成します:
+
+// Cat型は吠えないので、neverが返ります。
+type NeverCat = ExtractDogish<Cat>;
+// Wolf型は吠えるので、Wolf型が返ります。
```

Comment:
[IMO]

> A cat doesn't bark, so it will return never
> A wolf will bark, so it returns the wolf shape

この文脈では、 型としてではなく、動物として扱っているので、↓のようにしたほうが良いかなと思いました
がいかがでしょうか？

```suggestion
// 猫は吠えないので、neverが返ります。
type NeverCat = ExtractDogish<Cat>;
// 狼は吠えるので、狼の型 (Wolf) が返ります。
```

Reply:
型がついてた方が分かりやすいかと思いましたが、確かにですね